File: UI/src/main/java/ca/warp7/rt/java/core/feature/Feature.java
package ca.warp7.rt.java.core.feature;

import javafx.collections.ObservableList;
import javafx.scene.Parent;

public interface Feature {

    void init();

    ObservableList<FeatureAction> getActionList();

    String getFeatureId();

    Parent onAction(FeatureAction.Type type, String params);

    boolean onCloseRequest();
}

File: UI/src/main/kotlin/ca/warp7/rt/core/feature/Feature.kt
package ca.warp7.rt.core.feature

import javafx.scene.Parent

interface Feature {
    val link: FeatureLink
    val loadedTabs: List<FeatureItemTab>
    val featureId: String
    fun onOpenTab(tab: FeatureItemTab): Parent? = null
    fun onClose(): Boolean = true
    fun setFocused(focused: Boolean) = Unit
    fun onOpen(): Pair<Parent?, Parent?> = null to null
}
